plans/create.pp

```diff
@@ -0,0 +1,88 @@
+#
+# @summary Plan to provision a Virtual Machine in different cloud providers (GCP, AWS & Azure)
+#
+# @param provider
+#   The cloud provider to be provisioned
+#
+# @param resource_name
+#   The name of the resource to be provisioned
+#
+# @param instance_type
+#   The instance type to be provisioned
+#
+# @param image
+#   The cloud image that is used for new instance provisioning, format differs
+#   depending on provider
+#
+# @param region
+#   Which region to provision infrastructure in, if not provided default will
+#   be determined by provider
+#
+# @param ssh_key
+#   The SSH key name to be provisioned & access the machine
+#
+# @param ssh_user
+#   The ssh user to provision on provisioned VMs
+#
+# @param subnet
+#   Name or ID of an existing subnet to attach newly provisioned VMs
+#
+# @param security_group_ids
+#   The list of security group which will be attached to the newly provisioned VMs
+#
+# @param backend
+#   The backend to be provisioned
+#
+# @param bucket
+#   The bucket to be provisioned
+#
+# @param profile
+#   The name of the profile to be used for provisioning
+#
+# @param tags
+#  The list of tags to be attached to the newly provisioned VMs
+#
+plan provision::create(
+  String[1] $resource_name                                           = 'puppetlabs-provision',
+  Enum['gcp', 'aws', 'azure'] $provider                              = 'aws',
+  Enum['xlarge', 'large', 'medium', 'small', 'micro'] $instance_type = 'micro',
+  Enum['arm', 'amd', 'intel'] $architecture                          = 'intel',
+  String[1] $image                                                   = undef,
+  String[1] $region                                                  = undef,
```

Comment:
Where does the default region get specified?

Reply:
added default region to `us-west-2`